src/colmap/estimators/cost_functions.h

```diff
@@ -468,61 +457,67 @@ struct Point3DAlignmentCostFunctor
         EigenVector3Map<T>(b_from_a_translation);
     Eigen::Map<Eigen::Matrix<T, 3, 1>> residuals(residuals_ptr);
     residuals = point_in_b - point_in_b_prior_.cast<T>();
-    residuals.applyOnTheLeft(point_sqrt_info_in_b_prior_.template cast<T>());
     return true;
   }
 
  private:
   const Eigen::Vector3d point_in_b_prior_;
-  const Eigen::Matrix3d point_sqrt_info_in_b_prior_;
 };
 
-// A cost function that wraps another one and whiten its residuals with an
-// isotropic covariance, i.e. assuming that the variance is identical in and
-// independent between each dimension of the residual.
+template <typename... Args>
+auto LastValueParameterPack(Args&&... args) {
+  return std::get<sizeof...(Args) - 1>(std::forward_as_tuple(args...));
+}
+
+// A cost function that wraps another one and whitens its residuals with a given
+// left covariance. For example, to weight the reprojection error with a
+// image measurement covariance, one can wrap it as:
+//
+//    using ReprojCostFunctor = ReprojErrorCostFunctor<PinholeCameraModel>;
+//    ceres::CostFunction* cost_function =
+//        CovarianceWeightedCostFunctor<ReprojCostFunctor>::Create(
+//            point2D_cov, point2D));
 template <class CostFunctor>
-class IsotropicNoiseCostFunctorWrapper {
-  class LinearCostFunction : public ceres::CostFunction {
-   public:
-    explicit LinearCostFunction(const double s) : s_(s) {
-      set_num_residuals(1);
-      mutable_parameter_block_sizes()->push_back(1);
-    }
+class CovarianceWeightedCostFunctor {
+ public:
+  static constexpr int kNumResiduals = CostFunctor::kNumResiduals;
+  using kParameterDims = typename CostFunctor::kParameterDims;
 
-    bool Evaluate(double const* const* parameters,
-                  double* residuals,
-                  double** jacobians) const final {
-      *residuals = **parameters * s_;
-      if (jacobians && *jacobians) {
-        **jacobians = s_;
-      }
-      return true;
-    }
+  // Covariance or sqrt information matrix type.
+  using CovMat = Eigen::Matrix<double, kNumResiduals, kNumResiduals>;
+
+  template <typename... Args>
+  explicit CovarianceWeightedCostFunctor(const CovMat& left_cov, Args&&... args)
```

Comment:
I dont think ``left_cov`` is the right name. Rather it s ``left_sqrt_info`` if you want to force the "left" to be in the name.

Reply:
Right, I fixed this. Thanks.